lms/djangoapps/course_api/api.py

```diff
@@ -79,6 +81,13 @@ def course_detail(request, username, course_key):
         course_key,
     )
     overview.effective_user = user
+    if user.is_anonymous:
+        overview.certificate_available_date = None
+        return overview
+    current_status = certificate_status_for_student(user, course_key)
+
+    if not allow_certificate_available_date(overview, current_status):
```

Comment:
I think because this is not a user-specific API that we shouldn't be conditionally displaying this based on the state of a student's certificate. I think this check should actually be what `allow_certificate_available_date` is, but without `CertificateStatuses.is_passing_status(current_status['status'])`.

Reply:
I'm not following this completely, but we should avoid mixing together code that returns course metadata with code that returns student data.

Reply:
It's a good point by @rhysyngsun. I agree we should not check student states for `certificate_available_date` in the API.